sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSessionReceiverAsyncClient.java

```diff
@@ -0,0 +1,124 @@
+     */
+    public Mono<ServiceBusReceiverAsyncClient> acceptNextSession() {
+        return unNamedSessionManager.getActiveLink().flatMap(receiveLink -> receiveLink.getSessionId()
+            .map(sessionId -> {
+                ReceiverOptions newReceiverOptions = new ReceiverOptions(receiverOptions.getReceiveMode(),
+                    receiverOptions.getPrefetchCount(), sessionId, null, receiverOptions.getMaxLockRenewDuration());
+                final ServiceBusSessionManager sessionSpecificManager = new ServiceBusSessionManager(entityPath,
+                    entityType, connectionProcessor, tracerProvider, messageSerializer, newReceiverOptions,
+                    receiveLink);
+                return new ServiceBusReceiverAsyncClient(fullyQualifiedNamespace, entityPath,
+                    entityType, newReceiverOptions, connectionProcessor, ServiceBusConstants.OPERATION_TIMEOUT,
+                    tracerProvider, messageSerializer, onClientClose, sessionSpecificManager);
+            }));
+    }
+
+    /**
+     * Create a link for the "sessionId" and use the link to create a {@link ServiceBusReceiverAsyncClient}
+     * to receive messages from the session.
+     * @param sessionId The session Id.
+     * @return A {@link ServiceBusReceiverAsyncClient} that is tied to the specified session.
+     * @throws IllegalArgumentException if {@code sessionId} is null or empty.
+     */
+    public Mono<ServiceBusReceiverAsyncClient> acceptSession(String sessionId) {
+        if (CoreUtils.isNullOrEmpty(sessionId)) {
+            return monoError(logger, new IllegalArgumentException("sessionId can not be null or empty"));
+        }
+        ReceiverOptions newReceiverOptions = new ReceiverOptions(receiverOptions.getReceiveMode(),
+            receiverOptions.getPrefetchCount(), sessionId, null, receiverOptions.getMaxLockRenewDuration());
+        final ServiceBusSessionManager sessionSpecificManager = new ServiceBusSessionManager(entityPath, entityType,
+            connectionProcessor, tracerProvider, messageSerializer, newReceiverOptions);
+
+        return sessionSpecificManager.getActiveLink().thenReturn(new ServiceBusReceiverAsyncClient(
+            fullyQualifiedNamespace, entityPath, entityType, newReceiverOptions, connectionProcessor,
+            ServiceBusConstants.OPERATION_TIMEOUT, tracerProvider, messageSerializer, onClientClose,
+            sessionSpecificManager));
+    }
+
+    /**
+     * Create a {@link ServiceBusReceiverAsyncClient} that processes at most {@code maxConcurrentSessions} sessions.
+     *
+     * @param maxConcurrentSessions Maximum number of concurrent sessions to process at any given time.
+     *
+     * @return The {@link ServiceBusReceiverAsyncClient} object that will be used to receive messages.
+     * @throws IllegalArgumentException if {@code maxConcurrentSessions} is less than 1.
+     */
+    public ServiceBusReceiverAsyncClient getReceiverClient(int maxConcurrentSessions) {
```

Comment:
This API can be hidden for now until we decide that this is required. We can consider adding this post GA.

Reply:
Removed it.